politeiawww/codetracker/github/api/ratelimit.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"time"
+)
+
+const (
+	rateLimitURL = "https://api.github.com/rate_limit"
+)
+
+// RateLimit determines if the current rate limit for the client is about
+// to be tripped.  If so, it will go to sleep for a given perioud of time.
+func (a *Client) RateLimit() (RateLimitRule, error) {
+	defer a.rateLimitMtx.Unlock()
```

Comment:
Like I said earlier, make the mutex anon and put it as the first entry to the struct (alignment reasons). If you do that you can simply call `a.Lock()` and `a.Unlock()`. And for OCD reasons please put the lock in front of the unlock.